### PR TITLE
AzureMonitor: Make all query properties optional 

### PR DESCRIPTION
**Pull request by joshhunt**
**What this PR does / why we need it**:

This PR makes all properties and sub properties of `AzureMonitorQuery` optional, in preparation to removing Angular from Azure Monitor 🎈 

The old Azure Monitor angular query controller sets a big default query value (which isn't even correctly typed AFAICT). When moving this to React, I'm trying to not have have to set this big default value, and instead have every property be optional, and set defaults at the last minute where possible.

**Which issue(s) this PR fixes**:

Related to #33706

### CODE REVIEW

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts [hunk 1]
```diff
@@ -123,21 +126,19 @@ export default class AzureLogAnalyticsDatasource extends DataSourceWithBackend<
...
+
       azureLogAnalytics: {
         resultFormat: item.resultFormat,
-        query: query,
+        query,
         resource,
 
-        // TODO: Workspace is deprecated and should be migrated to Resources
-        workspace: workspace,
+        // Workspace is here for backwards compat
```
[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/azure_log_analytics/azure_log_analytics_datasource.ts:140]: good shout - its for pre grafana 8. we moved away from workspace in grafana 8.

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts [hunk 2]
```diff
@@ -109,50 +111,55 @@ export interface AzureMetricDimension {
 }
 
 export interface AzureMetricQuery {
-  resourceGroup: string | undefined;
-  resourceName: string | undefined;
-  metricDefinition: string | undefined;
-  metricNamespace: string | undefined;
-  metricName: string | undefined;
+  resourceGroup?: string | undefined;
```
Comment (sarahzinger) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts:114]: if you have a `?` do you need this `| undefined` as well?

**Review by sarahzinger — APPROVED** [review]

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/__mocks__/query.ts [hunk 3]
```diff
@@ -42,6 +42,5 @@ export default function createMockQuery(): AzureMonitorQuery {
     refId: 'A',
     subscription: '99999999-cccc-bbbb-aaaa-9106972f9572',
     subscriptions: ['99999999-cccc-bbbb-aaaa-9106972f9572'],
-    format: 'dunno lol', // unsure what this value should be. It's not there at runtime, but it's in the ts interface
```
[the remark this answers is not in the document]
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/__mocks__/query.ts:45]: I looked into it further and it wasnt actually used anywhere so i removed it!!!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts (hunk 2 above)
Comment (joshhunt) [reply public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts:114]: Good catch - nope!!!

File: public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts [hunk 4]
```diff
@@ -26,17 +26,19 @@ export enum AzureQueryType {
...
 
+// Actually, everything on here is optional because the object is empty when a new panel
+// is created
 export interface AzureMonitorQuery extends DataQuery {
-  queryType: AzureQueryType;
-  format: string;
-  subscription: string;
-  subscriptions: string[];
+  queryType?: AzureQueryType;
+  // format?: string; // this is not used by the backend at all, and isnt set anywhere??
```
Comment (andresmgot) [inline public/app/plugins/datasource/grafana-azure-monitor-datasource/types/index.ts:33]: this can be deleted, right?

**Review by andresmgot — COMMENTED** [review]
This leads to _lots_ of conditionals and also hide errors. For example, the compiler will tell me nothing if I don't set a `query` for AzureLogsQuery (and I send it) but that value should be mandatory. Can we differentiate between actual optional and mandatory fields?

**Review by andresmgot — APPROVED** [review]
After the offline conversation, I get the point that the query _can_ actually have some mandatory fields missing if the user has not added them (yet). I slightly prefer empty (zero) values rather than `undefined` values but that should not block this PR. Thanks!

Extra bonus points could be that the methods that actually perform the `query` verifies that all the required values are set.

**Pull request merged**